### ISSUE TITLE
Add "OldVersion" parameter to SampleConfigs json files

### ISSUE THREAD

**Issue by joekayaker** [Esri/arcgis-powershell-dsc#272]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.0.1

### Affected Resource(s)

SampleConfigs\v3\Base Deployment

* BaseDeployment-DualMachine.json
* BaseDeployment-MultiMachine.json
* BaseDeployment-SingleMachine.json
* BaseDeployment-SingleMachine-CloudStore.json
* BaseDeployment-ThreeMachine.json
* BaseDeployment-ThreeMachine-per-node-licenses.json
* BaseDeployment-ThreeMachine-ssl-per-Node-Example1.json
* BaseDeployment-ThreeMachine-ssl-per-Node-Example2.json

### Configuration Files

Excerpt from BaseDeployment-SingleMachine.json

```
"ConfigData": {
        "Version": "[Version of ArcGIS Deployment]",
        "ServerContext": "server",
        "PortalContext": "portal",
        "ServerRole": "GeneralPurposeServer",
        ....
```

### Expected Behavior

TL;DR: Should "OldVersion": "[version you are upgrading from]", be included after "Version" parameter in json files?

I have used the argis-powershell-dsc (ver. 2.0, 3.0) for deploying and upgrading Enterprise since version 10.6.
I just noticed that the new [3.0 wiki documentation](https://github.com/Esri/arcgis-powershell-dsc/wiki/V3.-Upgrade-an-ArcGIS-Enterprise-deployment) indicates, in the "Edit your configuration files" section, that upgrades should now include a parameter for "OldVersion". I assume this
is supposed to be in the "ConfigData" section of the json files. None of the SampleConfigs\v3\Base Deployment\*.json files have this parameter included.

Is this a "required" parameter at version 3.0 or just a suggested parameter?

I've never contributed to public repositories before. I'd like to add this to all the json files, but before I do, I wanted to confirm that it should be added and the "ConfigData" section is the correct location. Please confirm if this is correct and I'll add the parameter to the files and submit a pull request.

### Actual Behavior

Files do not contain "OldVersion" parameter.

### References

[Edit your configuration files](https://github.com/Esri/arcgis-powershell-dsc/wiki/V3.-Upgrade-an-ArcGIS-Enterprise-deployment) section of 3.0 wiki documentation.

**Comment by cameronkroeker** [comment]
Hello @joekayaker,

The **OldVersion** parameter is only required when performing `-Mode Upgrade`. It goes in the **ConfigData** block of the json file. For example:
```
"ConfigData":  {
                       "Version":  "10.8",
		       "OldVersion": "10.7.1"
}
```
It is referenced within the following wiki doc:

**OldVersion** - To be provided for upgrades only; this is the present version of the different components that will be upgraded.

https://github.com/Esri/arcgis-powershell-dsc/wiki/V3.-Variables-reference-page-for-JSON-configuration-files

Since this parameter is only required with upgrades we do not plan on adding it to the current json files within SampleConfigs\v3\Base Deployment, as these are samples for fresh deployments.

**Comment by joekayaker** [comment]
Thanks @cameronkroeker, that's great. It makes sense to not add it to the json files since it's only for upgrades. I thought it would get ignored based upon the "-mode" switch.
